File: app/src/main/java/org/mesonet/app/mesonetdata/MesonetData.java
package org.mesonet.app.mesonetdata;

import org.mesonet.app.formulas.UnitConverter;

import java.util.Date;
import java.util.Calendar;



public interface MesonetData
{
    Double GetTemp();
    Double GetApparentTemp();
    Double GetDewPoint();
    Double GetWindSpd();
    UnitConverter.CompassDirections GetWindDirection();
    Double GetMaxWind();
    Double Get24HrRain();
    Integer GetHumidity();
    Double GetPressure();
    Date GetTime ();
}

File: dataprocessing/src/main/java/org/mesonet/dataprocessing/site/mesonetdata/MesonetData.kt
package org.mesonet.dataprocessing.site.mesonetdata

import org.mesonet.dataprocessing.formulas.UnitConverter
import java.util.Date


interface MesonetData {
    fun GetTemp(): Double?
    fun GetApparentTemp(): Double?
    fun GetDewpoint(): Double?
    fun GetWindSpd(): Double?
    fun GetWindDirection(): UnitConverter.CompassDirections?
    fun GetMaxWind(): Double?
    fun Get24HrRain(): Double?
    fun GetHumidity(): Int?
    fun GetPressure(): Double?
    fun GetTime(): Date?
}
